sycl/doc/SYCLGraphUsageGuide.md

```diff
@@ -0,0 +1,493 @@
+  namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+  queue q{default_selector{}};
+
+  // Lifetime of buffers must exceed the lifetime of graphs they are used in.
+  buffer<T> bufferA{dataA.data(), range<1>{elements}};
+  bufferA.set_write_back(false);
+  buffer<T> bufferB{dataB.data(), range<1>{elements}};
+  bufferB.set_write_back(false);
+  buffer<T> bufferC{dataC.data(), range<1>{elements}};
+  bufferC.set_write_back(false);
+
+  {
+    // New object representing graph of command-groups
+    sycl_ext::command_graph graph(q.get_context(), q.get_device(),
+          {sycl_ext::property::graph::assume_buffer_outlives_graph{}});
+
+
+    // `q` will be put in the recording state where commands are recorded to
+    // `graph` rather than submitted for execution immediately.
+    graph.begin_recording(q);
+
+    // Record commands to `graph` with the following topology.
+    //
+    //      increment_kernel
+    //       /         \
+    //   A->/        A->\
+    //     /             \
+    //   add_kernel  subtract_kernel
+    //     \             /
+    //   B->\        C->/
+    //       \         /
+    //     decrement_kernel
+
+    q.submit([&](handler& cgh) {
+      auto pData = bufferA.get_access<access::mode::read_write>(cgh);
+      cgh.parallel_for<increment_kernel>(range<1>(elements),
+                                         [=](item<1> id) { pData[id]++; });
+    });
+
+    q.submit([&](handler& cgh) {
+      auto pData1 = bufferA.get_access<access::mode::read>(cgh);
+      auto pData2 = bufferB.get_access<access::mode::read_write>(cgh);
+      cgh.parallel_for<add_kernel>(range<1>(elements),
+                                   [=](item<1> id) { pData2[id] += pData1[id]; });
+    });
+
+    q.submit([&](handler& cgh) {
+      auto pData1 = bufferA.get_access<access::mode::read>(cgh);
+      auto pData2 = bufferC.get_access<access::mode::read_write>(cgh);
+      cgh.parallel_for<subtract_kernel>(
+          range<1>(elements), [=](item<1> id) { pData2[id] -= pData1[id]; });
+    });
+
+    q.submit([&](handler& cgh) {
+      auto pData1 = bufferB.get_access<access::mode::read_write>(cgh);
+      auto pData2 = bufferC.get_access<access::mode::read_write>(cgh);
+      cgh.parallel_for<decrement_kernel>(range<1>(elements), [=](item<1> id) {
+        pData1[id]--;
+        pData2[id]--;
+      });
+    });
+
+    // queue `q` will be returned to the executing state where commands are
+    // submitted immediately for extension.
+    graph.end_recording();
+
+    // Finalize the modifiable graph to create an executable graph that can be
+    // submitted for execution.
+    auto exec_graph = graph.finalize();
+
+    // Execute graph
+    q.submit([&](handler& cgh) {
+      cgh.ext_oneapi_graph(exec_graph);
+    }).wait();
+  }
+
+  // Check output using host accessors
+  host_accessor hostAccA(bufferA);
+  host_accessor hostAccB(bufferB);
+  host_accessor hostAccC(bufferC);
+
+  ...
+```
+
+### Dynamic Parameter Update
+
```

Comment:
If we can link to the dynamic parameter section of the spec here it might help the reader.

Reply:
I don't think it's possible to link to an asciidoc heading from markdown, as far as I can find anyway.